content/300-guides/150-testing/150-integration-testing.mdx

```diff
@@ -0,0 +1,450 @@
+import { createOrder, Customer, OrderInput } from '../src/functions/index'
+
+afterAll(async () => {
+  await prisma.$disconnect()
+})
+
+it('should create 1 new customer with 1 order', async () => {
+  // The new customers details
+  const customer: Customer = {
+    id: 2,
+    name: 'Hermione Granger',
+    email: 'hermione@hogwarts.io',
+    address: '2 Hampstead Heath',
+  }
+  // The new orders details
+  const order: OrderInput = {
+    customer,
+    productId: 1,
+    quantity: 1,
+  }
+
+  // Create the order and customer
+  await createOrder(order)
+
+  // Check if the new customer was created by filtering on unique email field
+  const newCustomer = await prisma.customer.findUnique({
+    where: {
+      email: customer.email,
+    },
+  })
+
+  // Check if the new order was created by filtering on unique email field of the customer
+  const newOrder = await prisma.order.findFirst({
+    where: {
+      customer: {
+        email: customer.email,
+      },
+    },
+  })
+
+  // Expect the new customer to have been created and match the input
+  expect(newCustomer).toEqual(customer)
+  // Expect the new order to have been created and contain the new customer
+  expect(newOrder).toHaveProperty('customerId', 2)
+})
+
+it('should create 1 order with an existing customer', async () => {
+  // The existing customers email
+  const customer: Customer = {
+    email: 'harry@hogwarts.io',
+  }
+  // The new orders details
+  const order: OrderInput = {
+    customer,
+    productId: 1,
+    quantity: 1,
+  }
+
+  // Create the order and connect the existing customer
+  await createOrder(order)
+
+  // Check if the new order was created by filtering on unique email field of the customer
+  const newOrder = await prisma.order.findFirst({
+    where: {
+      customer: {
+        email: customer.email,
+      },
+    },
+  })
+
+  // Expect the new order to have been created and contain the existing customer with an id of 1 (Harry Potter from the seed script)
+  expect(newOrder).toHaveProperty('customerId', 1)
+})
+```
+
+## Running the tests
+
+The idea with this setup is to isolate a real world scenario so that you can test your applications functionality against real data in a controlled environment.
+
+To that end you have setup a Docker container which will be home to a Postgres database. This database will be seeded with some data, have tests run against it and then be destroyed. You can set this all up via some scripts in your projects `package.json` file.
+
+```json file=package.json
+  "scripts": {
+    "migrate": "npx prisma migrate dev --name init",
+    "docker:up": "docker-compose up -d",
+    "docker:down": "docker compose down",
+    "seed": "node --loader ts-node/esm ./prisma/seed.ts",
+    "bootstrap": "yarn docker:up && yarn migrate && yarn seed",
+    "test": "yarn bootstrap && jest -i && yarn docker:down"
+  },
+```
+
+To run create the container and run the test you would run the `test` script. Breaking this down:
```

Comment:
Confused about this sentence: To run create the container and run the test you would run the `test` script. Should it be "To start the test, run the `test` script" and if so can we add:

```
npm run test
```

Would replace 'breaking this down' to something more formal - "The `seed` script runs the following steps:"